integration-cli/docker_api_network_test.go

```diff
@@ -41,6 +41,31 @@ func (s *DockerSuite) TestApiNetworkCreateDelete(c *check.C) {
 	c.Assert(isNetworkAvailable(c, name), checker.Equals, false)
 }
 
+func (s *DockerSuite) TestApiNetworkCreateCheckDuplicate(c *check.C) {
+	name := "testcheckduplicate"
+	configOnCheck := types.NetworkCreate{
+		Name:           name,
+		CheckDuplicate: true,
+	}
+	configNotCheck := types.NetworkCreate{
+		Name:           name,
+		CheckDuplicate: false,
+	}
+	// First check whether the network already exists or not
+	if isNetworkAvailable(c, name) {
+		// Creating another network with same name and CheckDuplicate must fail
+		createNetwork(c, configOnCheck, false)
```

Comment:
isn't this covered above? https://github.com/docker/docker/pull/18767/files#diff-d27869c471324416cf8914063c3ae8c3R36 (not sure)